spec/rubocop/cop/lint/literal_in_interpolation_spec.rb

```diff
@@ -507,5 +507,16 @@
         RUBY
       end
     end
+
+    context 'with invalid string literal' do
+      it 'registers an offense' do
+        expect_offense(<<~'RUBY')
+          "#{"\201\203"}"
+             ^^^^^^^^^^ Literal interpolation detected.
+        RUBY
+
+        expect_no_corrections
```

Comment:
I think the following autocorrection behavior is expected.

```suggestion
        expect_correction(<<~RUBY)
          "\201\203"
        RUBY
```

Reply:
I'm not sure why, but I get the following error

```
Cannot extract source from uninitialized Source::Buffer
     # ./lib/rubocop/cop/corrector.rb:34:in `initialize'
     # ./lib/rubocop/rspec/cop_helper.rb:80:in `new'
     # ./lib/rubocop/rspec/cop_helper.rb:80:in `_investigate'
     # ./lib/rubocop/rspec/expect_offense.rb:166:in `block in expect_correction'
     # ./lib/rubocop/rspec/expect_offense.rb:152:in `expect_correction'
     # ./spec/rubocop/cop/lint/literal_in_interpolation_spec.rb:518:in `block (4 levels) in <top (required)>'
```

when this patch is applied 

```
diff --git a/lib/rubocop/cop/lint/literal_in_interpolation.rb b/lib/rubocop/cop/lint/literal_in_interpolation.rb
index 2cbaa2231..dd1f32d5e 100644
--- a/lib/rubocop/cop/lint/literal_in_interpolation.rb
+++ b/lib/rubocop/cop/lint/literal_in_interpolation.rb
@@ -35,7 +35,6 @@ module RuboCop
           return if in_array_percent_literal?(begin_node) && /\s|\A\z/.match?(expanded_value)
 
           add_offense(final_node) do |corrector|
-            next if final_node.str_type? && !final_node.value.valid_encoding?
             next if final_node.dstr_type? # nested, fixed in next iteration
 
             corrector.replace(final_node.parent, expanded_value)
diff --git a/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb b/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb
index a47fe769f..b0add5196 100644
--- a/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb
+++ b/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb
@@ -515,7 +515,9 @@ RSpec.describe RuboCop::Cop::Lint::LiteralInInterpolation, :config do
              ^^^^^^^^^^ Literal interpolation detected.
         RUBY
 
-        expect_no_corrections
+        expect_correction(<<~'RUBY')
+          "\201\203"
+        RUBY
       end
     end
   end
```

Reply:
It may require some refactoring, but how about the following approach?

```diff
diff --git a/lib/rubocop/cop/lint/literal_in_interpolation.rb b/lib/rubocop/cop/lint/literal_in_interpolation.rb
index 2cbaa2231..7c0fbd875 100644
--- a/lib/rubocop/cop/lint/literal_in_interpolation.rb
+++ b/lib/rubocop/cop/lint/literal_in_interpolation.rb
@@ -35,10 +35,15 @@ module RuboCop
           return if in_array_percent_literal?(begin_node) && /\s|\A\z/.match?(expanded_value)

           add_offense(final_node) do |corrector|
-            next if final_node.str_type? && !final_node.value.valid_encoding?
             next if final_node.dstr_type? # nested, fixed in next iteration

-            corrector.replace(final_node.parent, expanded_value)
+            replacement = if final_node.str_type? && !final_node.value.valid_encoding?
+                            final_node.source.delete_prefix('"').delete_suffix('"')
+                          else
+                            expanded_value
+                          end
+
+            corrector.replace(final_node.parent, replacement)
           end
         end

diff --git a/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb b/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb
index a47fe769f..b0add5196 100644
--- a/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb
+++ b/spec/rubocop/cop/lint/literal_in_interpolation_spec.rb
@@ -515,7 +515,9 @@ RSpec.describe RuboCop::Cop::Lint::LiteralInInterpolation, :config do
              ^^^^^^^^^^ Literal interpolation detected.
         RUBY

-        expect_no_corrections
+        expect_correction(<<~'RUBY')
+          "\201\203"
+        RUBY
       end
     end
   end
```

Reply:
Awesome, thanks!